### PR TITLE
Don't export the indicators dependency

### PR DESCRIPTION
**Pull request by tobim**
Building plugins against an installed VAST was broken. This is the fix.

### :memo: Checklist

- [ ] All user-facing changes have changelog entries.
- [ ] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

Try building a plugin against an installed VAST, see that it fails without this commit.

### CODE REVIEW

File: libvast/CMakeLists.txt [hunk 1]
```diff
@@ -533,7 +533,16 @@ add_feature_info("VAST_ENABLE_BUNDLED_INDICATORS"
...
+    # CMake exports private interface dependencies because they might depend on
+    # other non-interface libraries. CMake could scan the depency tree to figure
+    # out when this is actually needed, but it doesn't.
+    # We know that indicators itself doesn't have any dependency, so we work
+    # around this behavior by not linking to the target.
+    # https://gitlab.kitware.com/cmake/cmake/-/issues/15415 for more discussion.
+    get_target_property(indicators_include_directory indicators::indicators
+                        INTERFACE_INCLUDE_DIRECTORIES)
+    target_include_directories(
+      libvast_native_plugins PRIVATE "${indicators_include_directory}/include")
```
Comment (6yozo) [inline libvast/CMakeLists.txt:545]: I cannot tell exactly what this change is doing, but it builds fine, so it must be ok.

File: tools/vast-regenerate/main.cpp [hunk 2]
```diff
@@ -93,10 +93,8 @@ caf::error write_index_bin(const std::vector<vast::uuid>& uuids,
   for (const auto& uuid : uuids) {
     if (auto uuid_fb = pack(builder, uuid))
       partition_offsets.push_back(*uuid_fb);
-    else {
-      fmt::print(stderr, "failed to pack uuid {}: {}\n", uuid, uuid_fb.error());
-      return 1;
-    }
+    else
+      return uuid_fb.error();
```
Comment (6yozo) [inline tools/vast-regenerate/main.cpp:97]: We are losing the context here("failed to pack uuid"), but the proper error message is returned to the caller; this is ok as it is or maybe:

```
      return caf::make_error(vast::ec::convert_error,
                             fmt::format("failed to pack uuid {}", uuid),
                             uuid_fb.error());
```

File: tools/vast-regenerate/main.cpp [hunk 3]
```diff
@@ -284,7 +282,7 @@ with an incorrect all-zero event count.
       path = arg;
     }
   }
-  if (!path) {
+  if (path.empty()) {
```
Comment (6yozo) [inline tools/vast-regenerate/main.cpp:285]: Nice! Now it can detect when the path is not given in CLI.

**Review by 6yozo — APPROVED** [review]
The PR is good to go, as long as my understanding of CMAKE goes.

File: tools/vast-regenerate/main.cpp (hunk 2 above)
Comment (tobim) [reply tools/vast-regenerate/main.cpp:97]: All `pack` overloads share `caf::error` as return type. The one for `uuid` can't fail though, and I don't see how that would change in the future.

**Pull request merged**